fuel-block-executor/src/refs/contract.rs

```diff
@@ -0,0 +1,75 @@
+use fuel_core_interfaces::{
+    common::{
+        fuel_storage::{
+            Mappable,
+            MerkleRootStorage,
+            StorageAsRef,
+            StorageInspect,
+        },
+        fuel_tx::{
+            Bytes32,
+            ContractId,
+        },
+    },
+    db::{
+        ContractsAssets,
+        ContractsLatestUtxo,
+        ContractsState,
+    },
+};
+use std::borrow::Cow;
+
+/// The wrapper around `contract_id` to simplify work with `Contract` in the database.
+pub struct ContractRef<'a, Database> {
+    database: Database,
+    contract_id: &'a ContractId,
```

Comment:
do we really need to lifetime reference this? the ContractId implements copy and should be fairly cheap to own here 🤔

Reply:
I added this type to simplify usage without affecting performance. Because the type is public and used inside of another crate, the compiler will not optimize usage of this struct, and it will add a smaaaaaaaaaaaaaall overhead(to copy 32 bytes)=)

If this lifetime causes pain to your eyes, I will remove it=D

Reply:
While copying does incur a small overhead, dereferencing is also not free from a cache standpoint. I also think the simpler struct definition we get may be worth the cost of copying the Bytes32 .

Reply:
I don't know why we are afraid of lifetimes=) But if two people don't like it, I will replace it with four copies.

Reply:
I know it's more efficient to use the lifetimes, but my preference is to try and keep the code as simple as possible until it's proven to be a bottleneck.

Reply:
Rather than a question of lifetimes, I think (if I'm understanding Brandon's original comment correctly) it's more of a question about when to use references vs. when to use copies. Personally, I think IDs (similar to ints) are something you should be able to copy without much fear of overhead. But I can also understand that in a performance-critical application, avoid copies is a general goal (but there is the potential cost of accumulating cache misses if this is in any sort of loop).